File: python-examples/browser-sdk-showcase/api/click_until_exhausted.py
from playwright.async_api import Page
from typing import TypedDict
from intuned_browser import click_until_exhausted
from intuned_browser import go_to_url

class Params(TypedDict):
        pass

async def automation(page: Page, params: Params | None = None, **_kwargs):
    await go_to_url(page, "https://careers.qualcomm.com/careers")
    button_locator = page.locator("xpath=//button[@class='btn btn-sm btn-secondary show-more-positions']")
    # Click until exhausted
    await click_until_exhausted(page=page, button_locator=button_locator, max_clicks=5)
    # Now all content is loaded and visible
    return "Success"

File: typescript-examples/browser-sdk-showcase/api/click-until-exhausted.ts
import { BrowserContext, Page } from "playwright";
import { clickUntilExhausted, goToUrl } from "@intuned/browser";

interface Params {
  // No params needed
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  await goToUrl({ page, url: "https://careers.qualcomm.com/careers" });

  const buttonLocator = page.locator(
    "xpath=//button[@class='btn btn-sm btn-secondary show-more-positions']"
  );

  // Click on the button to load more content 5 times.
  // Check https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/clickUntilExhausted for more details.
  await clickUntilExhausted({
    page,
    buttonLocator,
    maxClicks: 5,
  });

  // Now all content is loaded and visible
  return "Success";
}

